content/en/docs/tasks/traffic-management/ingress/secure-ingress/index.md

```diff
@@ -336,43 +491,90 @@ retrieves unique credentials corresponding to a specific `credentialName`.
 
 ### Configure a mutual TLS ingress gateway
 
-You can extend your gateway's definition to support
-[mutual TLS](https://en.wikipedia.org/wiki/Mutual_authentication). Change
-the credentials of the ingress gateway by deleting its secret and creating a new one.
-The server uses the CA certificate to verify
-its clients, and we must use the name `cacert` to hold the CA certificate.
+You can extend your gateway's definition to support [mutual TLS](https://en.wikipedia.org/wiki/Mutual_authentication).
+
+1. Change the credentials of the ingress gateway by deleting its secret and creating a new one.
+    The server uses the CA certificate to verify its clients, and we must use the name `cacert` to hold the CA certificate.
+
+    {{< text bash >}}
+    $ kubectl -n istio-system delete secret httpbin-credential
+    $ kubectl create -n istio-system secret generic httpbin-credential \
+      --from-file=tls.key=example_certs1/httpbin.example.com.key \
+      --from-file=tls.crt=example_certs1/httpbin.example.com.crt \
+      --from-file=ca.crt=example_certs1/example.com.crt
+    {{< /text >}}
+
+1. Configure the ingress gateway:
+
+{{< tabset category-name="config-api" >}}
+
+{{< tab name="Istio classic" category-value="istio-classic" >}}
+
+Change the gateway's definition to set the TLS mode to `MUTUAL`.
 
 {{< text bash >}}
-$ kubectl -n istio-system delete secret httpbin-credential
-$ kubectl create -n istio-system secret generic httpbin-credential --from-file=tls.key=httpbin.example.com.key \
---from-file=tls.crt=httpbin.example.com.crt --from-file=ca.crt=example.com.crt
+$ cat <<EOF | kubectl apply -f -
+apiVersion: networking.istio.io/v1alpha3
+kind: Gateway
+metadata:
+  name: mygateway
+spec:
+  selector:
+    istio: ingressgateway # use istio default ingress gateway
+  servers:
+  - port:
+      number: 443
+      name: https
+      protocol: HTTPS
+    tls:
+      mode: MUTUAL
+      credentialName: httpbin-credential # must be the same as secret
+    hosts:
+    - httpbin.example.com
+EOF
 {{< /text >}}
 
-1. Change the gateway's definition to set the TLS mode to `MUTUAL`.
+{{< /tab >}}
 
-    {{< text bash >}}
-    $ cat <<EOF | kubectl apply -f -
-    apiVersion: networking.istio.io/v1alpha3
-    kind: Gateway
-    metadata:
-     name: mygateway
-    spec:
-     selector:
-       istio: ingressgateway # use istio default ingress gateway
-     servers:
-     - port:
-         number: 443
-         name: https
-         protocol: HTTPS
-       tls:
-         mode: MUTUAL
-         credentialName: httpbin-credential # must be the same as secret
-       hosts:
-       - httpbin.example.com
-    EOF
-    {{< /text >}}
+{{< tab name="Gateway API" category-value="gateway-api" >}}
+
+A [Kubernetes Gateway](https://gateway-api.sigs.k8s.io/references/spec/#gateway.networking.k8s.io%2fv1beta1.Gateway)
+performing mutual TLS termination is configured no differently from one using simple TLS. The only difference
```

Comment:
I wonder if "mutual TLS termination is configured no differently" is going to be an issue... tools like cert-manager may always stick the CA cert in the secret?

Reply:
Good question, it certainly surprised me.

Reply:
Actually are you sure its true? https://github.com/istio/istio/blob/1c70653e0eec55aed47674590767cd2dd4e8ff64/pilot/pkg/config/kube/gateway/conversion.go#L1611

Istio cannot (or doesnt, anyways) tell what is in the secret at config gen time anyways... I think its only SIMPLE. I suspect the example would pass even if you didn't send client cert

Reply:
> I suspect the example would pass even if you didn't send client cert

No, it does fail without the client cert. Look at step 3, here: https://deploy-preview-12051--preliminary-istio.netlify.app/latest/docs/tasks/traffic-management/ingress/secure-ingress/#configure-a-mutual-tls-ingress-gateway

Reply:
:confused: I don't know how, but there is some bug somewhere I think...

Reply:
Oops, actually you are right. The failing curl command has the wrong value for cacert (`--cacert example.com.crt` should be `--cacert example_certs1/example.com.crt`), so it's failing for an unrelated reason.

So, Gateway API doesn't support mTLS currently? That seems like a serious omission.

Reply:
Yes, not currently

Reply:
How about we add an Istio option for now? Something like this:
```
apiVersion: gateway.networking.k8s.io/v1beta1
kind: Gateway
metadata:
  name: mygateway
  namespace: istio-system
spec:
  gatewayClassName: istio
  listeners:
  - name: https
    hostname: "httpbin.example.com"
    port: 443
    protocol: HTTPS
    tls:
      mode: Terminate
      certificateRefs:
      - name: httpbin-credential
      options:
        networking.istio.io/tls.mode: MUTUAL
    allowedRoutes:
      namespaces:
        from: Selector
        selector:
          matchLabels:
            kubernetes.io/metadata.name: default
```